strawberry/field.py

```diff
@@ -112,11 +84,9 @@ def field(
 
     field_definition = FieldDefinition(
         origin_name=None,  # modified by resolver in __call__
-        name=name,  # modified by resolver in __call__
-        type=None,  # type: ignore
-        origin=resolver,  # type: ignore
```

Comment:
We were leaving `origin` as `None` unless a resolver was provided. Was this correct? Would a resolver-less field never have an origin?

Reply:
we use origin here: https://github.com/strawberry-graphql/strawberry/blob/23ee3adce2bf062fb76080030c74d6218bcf4cf5/strawberry/types/type_resolver.py#L67 

a resolver-less field should have origin equal to the class where it was defined (might not be the case now) :)

Reply:
It ends up being set [here](https://github.com/strawberry-graphql/strawberry/blob/6ebbc76da0bf213256ad5695e1a370e00f011abf/strawberry/types/type_resolver.py#L366). I think my changes make sense*, as previously the constructor to `FieldDefinition` was _always_ being set to `None` or `resolver`, and then re-set to `resolver` in `__call__`. Now I don't set it in the initial constructor call and just defer it to the two other places.

*for now; at least until `FieldDefinition` goes by the wayside with the future overhaul of `StrawberryField`